### PR TITLE
ci: release providers

### PR DESCRIPTION
**Pull request by rvolosatovs**
## Feature or Problem

build and release PARs in CI. Publish `main` and tag builds to GHCR and AzureCR.

Example tag build available at https://github.com/wasmCloud/wasmCloud/actions/runs/8362687944 (`provider-blobstore-fs-v0.6.0-rc.1`)

https://github.com/wasmCloud/wasmCloud/pkgs/container/blobstore-fs/193605679?tag=0.6.0-rc.1

### CODE REVIEW

File: .github/workflows/wasmcloud.yml
Comment (thomastaylor312) [inline .github/workflows/wasmcloud.yml:309]: This should be `WASMCLOUD_ACCOUNT_OFFICIAL` to pick up on the org level secret that contains the issuer

File: .github/workflows/provider.yml
Comment (thomastaylor312) [inline .github/workflows/provider.yml:66]: Nit: Might just be easier to have these be set as env variables for this step rather than exporting in the script

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]

**Comment by thomastaylor312** [comment]
I've added all the necessary secrets I believe. So this should be good to go once we merge

File: .github/workflows/provider.yml
Comment (rvolosatovs) [reply .github/workflows/provider.yml:66]: That was my initial approach as well, however having these set to empty string breaks `wash` unfortunately, so the only way to support key autogeneration here is to check for the value explicitly before setting the env

Comment (thomastaylor312) [reply .github/workflows/provider.yml:66]: We wouldn’t want it to autogenerate keys though if they’re not set, right?

**Review by thomastaylor312 — APPROVED** [review]
Good to go but one more clarifying question I left in a previous thread

File: .github/workflows/provider.yml
Comment (rvolosatovs) [reply .github/workflows/provider.yml:66]: That would prevent this from running on forks, meaning that this would never run on PRs from forks. I do think that we want to build PARs always, at least for the convenience of testing proposed changes, but also to ensure a PR does not break this process

Comment (thomastaylor312) [reply .github/workflows/provider.yml:66]: Yeah I can definitely get behind that

**Pull request merged**
